packages/viewer/src/components/viewer-hit-result-indicator/viewer-hit-result-indicator.tsx

```diff
@@ -0,0 +1,254 @@
+    oldViewer?: HTMLVertexViewerElement
+  ): void {
+    oldViewer?.removeEventListener('frameDrawn', this.handleViewerFrameDrawn);
+    oldViewer?.removeEventListener(
+      'dimensionschange',
+      this.handleViewerDimensionsChange
+    );
+    newViewer?.addEventListener('frameDrawn', this.handleViewerFrameDrawn);
+    newViewer?.addEventListener(
+      'dimensionschange',
+      this.handleViewerDimensionsChange
+    );
+  }
+
+  /**
+   * @ignore
+   */
+  @Watch('position')
+  protected handlePositionChanged(
+    newPosition?: Vector3.Vector3,
+    oldPosition?: Vector3.Vector3
+  ): void {
+    this.transform = this.createTransform();
+
+    console.debug(
+      `Updating indicator position [previous=${JSON.stringify(
+        oldPosition
+      )}, current=${JSON.stringify(newPosition)}]`
+    );
+
+    this.indicator?.updateTransformAndNormal(this.transform, this.normal);
+  }
+
+  @Watch('normal')
+  protected handleNormalChanged(
+    newNormal?: Vector3.Vector3,
+    oldNormal?: Vector3.Vector3
+  ): void {
+    console.debug(
+      `Updating indicator normal [previous=${JSON.stringify(
+        oldNormal
+      )}, current=${JSON.stringify(newNormal)}]`
+    );
+
+    if (newNormal) {
+      this.indicator?.updateTransformAndNormal(this.transform, this.normal);
+    }
+  }
+
+  public render(): h.JSX.IntrinsicElements {
+    return (
+      <Host>
+        <canvas
+          ref={(el) => {
+            this.canvasRef = el;
+          }}
+          class="indicator"
+          width={this.viewer?.viewport.width}
+          height={this.viewer?.viewport.height}
+        />
+      </Host>
+    );
+  }
+
+  private handleViewerFrameDrawn = (): void => {
+    this.updatePropsFromViewer();
+  };
+
+  private handleViewerDimensionsChange = (): void => {
+    writeDOM(() => {
+      if (this.viewer != null && this.canvasRef != null) {
+        this.canvasRef.width = this.viewer.viewport.width;
+        this.canvasRef.height = this.viewer.viewport.height;
+
+        this.updateAndRedrawIndicator();
+      }
+    });
+  };
+
+  private handleResize = (): void => {
+    if (this.canvasRef != null) {
+      this.updateAndRedrawIndicator();
+    }
+  };
+
+  private updatePropsFromViewer = (): void => {
+    const { frame } = this.viewer || {};
+
+    if (frame != null) {
+      const widget = this.getIndicator();
+
+      widget.updateFrame(frame);
+    }
+  };
+
+  private setupIndicator = (canvasRef: HTMLCanvasElement): HitIndicator => {
+    console.debug(
+      `Initializing hit indicator. [initial-position=${JSON.stringify(
+        this.position
+      )}, initial-normal=${JSON.stringify(this.normal)} has-initial-frame=${
+        this.viewer?.frame != null
+      }]`
+    );
+
+    this.indicator = new HitIndicator(
+      canvasRef,
+      {
+        arrow: this.arrowColor,
+        plane: this.planeColor,
+      },
+      {
+        plane: this.planeOpacity,
+      }
+    );
+
+    if (this.position != null) {
```

Comment:
In what cases do we expect the position to be null? Should the `else` condition here include a warning or error?